ovos_config/base.py

```diff
@@ -0,0 +1,272 @@
+                    with open(path) as f:
+                        config = yaml.safe_load(f)
+                else:
+                    config = load_commented_json(path)
+                for key in config:
+                    self.__setitem__(key, config[key])
+                LOG.debug(f"Configuration {path} loaded")
+            except Exception as e:
+                LOG.exception(f"Error loading configuration '{path}'")
+        else:
+            LOG.debug(f"Configuration '{path}' not defined, skipping")
+
+    def reload(self):
+        self.load_local(self.path)
+
+    def store(self, path=None):
+        """Cache the received settings locally.
+
+        The cache will be used if the remote is unreachable to load settings
+        that are as close to the user's as possible.
+        """
+        path = path or self.path
+        if not path:
+            LOG.error(f"in memory configuration, no save location")
+            return
+        if self._get_file_format(path) == "yaml":
+            with open(path, 'w') as f:
+                yaml.dump(dict(self), f, allow_unicode=True,
+                          default_flow_style=False, sort_keys=False)
+        else:
+            with open(path, 'w') as f:
+                json.dump(self, f, indent=2)
+
+    def merge(self, conf):
+        merge_dict(self, conf)
+
+
+class ReadOnlyConfig(LocalConf):
+    """ read only  """
+
+    def __init__(self, path, allow_overwrite=False):
+        super().__init__(path)
+        self.allow_overwrite = allow_overwrite
+
+    def reload(self):
+        old = self.allow_overwrite
+        self.allow_overwrite = True
+        super().reload()
+        self.allow_overwrite = old
+
+    def __setitem__(self, key, value):
+        if not self.allow_overwrite:
+            raise PermissionError(f"{self.path} is read only! it can not be modified at runtime")
+        super().__setitem__(key, value)
+
+    def __setattr__(self, key, value):
+        if not self.allow_overwrite:
+            raise PermissionError(f"{self.path} is read only! it can not be modified at runtime")
+        super().__setattr__(key, value)
+
+    def merge(self, *args, **kwargs):
+        if not self.allow_overwrite:
+            raise PermissionError(f"{self.path} is read only! it can not be modified at runtime")
+        super().merge(*args, **kwargs)
+
+    def store(self, path=None):
+        if not self.allow_overwrite:
+            raise PermissionError(f"{self.path} is read only! it can not be modified at runtime")
+        super().store(path)
+
+
+class MycroftDefaultConfig(ReadOnlyConfig):
+    def __init__(self):
+        super().__init__(DEFAULT_CONFIG)
+
+    def set_root_config_path(self, root_config):
+        # in case we got it wrong / non standard
+        self.path = root_config
+        self.reload()
+
+
+class MycroftSystemConfig(ReadOnlyConfig):
+    def __init__(self, allow_overwrite=False):
+        super().__init__(SYSTEM_CONFIG, allow_overwrite)
```

Comment:
this flag was meant to be integrated into some mechanism providing root access, since sudo is needed for modifying this file